### DISCUSSION TITLE
irrational "fractions"

### DISCUSSION THREAD

**Discussion by nschloe** [Textualize/textual#122] [Ideas]
In textual grids, one can specify `fraction`s for sizing the columns:
```python
grid = await self.view.dock_grid(edge="left")
grid.add_column(fraction=8, name="left")
grid.add_column(fraction=13, name="right")
```
Do the fractions really need to be integers? I'd love to use irrational fractions, e.g., [the golden ratio](https://en.wikipedia.org/wiki/Golden_ratio).

**Comment by willmcgugan** [comment]
How would you want to express the irrational fraction in Python?

**Reply by nschloe** [reply]
Just floats.

**Reply by willmcgugan** [reply]
I suspect it should work. It might be typed as int. You might want to check it. If it doesn't work, it is probably an easy fix to make floats work.

**Reply by nschloe** [reply]
Right now I have
```python
            grid.add_column(fraction=8, name="left")
            grid.add_column(fraction=13, name="right")
```
(as an approximation to the golden ratio. Changing to
```python
            grid.add_column(fraction=8 / 13, name="left")
            grid.add_column(fraction=1, name="right")
```
gives the error
```
TypeError: both arguments should be Rational instances
```

**Reply by willmcgugan** [reply]
Ah. That's coming from the Fractions module. Next time I'm in that code I'll see if it could be modified to use floats.
